### PR TITLE
Feature/unit testing

### PR DESCRIPTION
**Pull request by lee-benjamin**
at long, long last.

### CODE REVIEW

File: src/main/java/com/benjamin/benleethium/resources/BenLeethiumResource.java [hunk 1]
```diff
@@ -55,7 +56,7 @@ public Response getHomeTimeline() {
     @Timed
     public Response updateStatus(@FormParam("message") String message) {
         try {
-            Status status = twitterInstance.getInstance().updateStatus(message);
+            Status status = this.twitterService.updateStatus(message);
```
Comment (kev-chan) [inline src/main/java/com/benjamin/benleethium/resources/BenLeethiumResource.java]: you've got twitterService.getHometimeline() above and this.twitterService.udpateStatus here...let's make it consistent. I'd recommend just 'twitterService'

File: src/main/java/com/benjamin/benleethium/resources/BenLeethiumResource.java [hunk 2]
```diff
@@ -77,7 +78,7 @@ public Response updateStatus(@FormParam("message") String message) {
     @Timed
     public Response searchHomeTimeline(@QueryParam("keyword") String keyword) {
         try {
-            List<Status> searchResults = twitterInstance.getInstance().searchHomeTimeline(keyword);
+            List<Status> searchResults = this.twitterService.searchHomeTimeline(keyword);
```
Comment (kev-chan) [inline src/main/java/com/benjamin/benleethium/resources/BenLeethiumResource.java]: same as above

File: src/test/java/com/benjamin/benleethium/resource/BenLeethiumResourceTest.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+
+public class BenLeethiumResourceTest {
+
+    private TwitterService twitterService;
+    private BenLeethiumResource benLeethiumResource;
+
+    @Before
+    public void setUp() {
+        twitterService = Mockito.mock(TwitterService.class);
```
Comment (kev-chan) [inline src/test/java/com/benjamin/benleethium/resource/BenLeethiumResourceTest.java]: Mockito.mock is fine. More of an FYI that you'll likely see just 'mock(TwitterService.class)' in most tests

File: src/test/java/com/benjamin/benleethium/resource/BenLeethiumResourceTest.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    @Test
+    public void testUpdateStatus() throws TwitterException {
+        String tweet = "A tweet a day doth a healthy bird make.";
+        String name = "Ben";
+        String screenName = "BenLeethium";
+        String profileImageURL = "ben.com";
+        Date date = new Date();
+
+        User parsedUser; // stripped version of a twitter4j.User
```
Comment (kev-chan) [inline src/test/java/com/benjamin/benleethium/resource/BenLeethiumResourceTest.java]: i would go ahead and just say 'User parsedUser = new User..."

File: src/test/java/com/benjamin/benleethium/resource/BenLeethiumResourceTest.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    @Test
+    public void testSearchHomeTimeline() throws TwitterException {
+        String[] tweets = {"This is a test.", "Tweeting my cares away.","A tweet a day doth a healthy bird make."};
+        String query = "tweet"; // No search will actually occur during this test
+        String name = "Ben";
+        String screenName = "BenLeethium";
+        String profileImageURL = "ben.com";
+        Date date = new Date();
+
+        // TwitterService.getHomeTimeline() returns a List<Status>
```
Comment (kev-chan) [inline src/test/java/com/benjamin/benleethium/resource/BenLeethiumResourceTest.java]: comment seems redundant as you can look at getTimeline method to infer the return type

File: src/main/java/com/benjamin/benleethium/resources/BenLeethiumResource.java (hunk 1 above)
Comment (lee-benjamin) [reply src/main/java/com/benjamin/benleethium/resources/BenLeethiumResource.java]: good catch

File: src/test/java/com/benjamin/benleethium/resource/BenLeethiumResourceTest.java (hunk 4 above)
Comment (lee-benjamin) [reply src/test/java/com/benjamin/benleethium/resource/BenLeethiumResourceTest.java]: whoops, left over from boilerplate code

File: src/test/java/com/benjamin/benleethium/resource/BenLeethiumResourceTest.java (hunk 5 above)
Comment (lee-benjamin) [reply src/test/java/com/benjamin/benleethium/resource/BenLeethiumResourceTest.java]: also left over from boilerplate at a time when there were two lists of diff types with diff values inside. I removed the comment.

**Pull request merged**